tests/python/test_dmatrix.py

```diff
@@ -207,6 +211,22 @@ def test_feature_names(self):
             with pytest.raises(ValueError):
                 bst.predict(dm)
 
+    @pytest.mark.skipif(**tm.no_pandas())
+    def test_save_binary(self):
+        import pandas as pd
+        with tempfile.TemporaryDirectory() as tmpdir:
+            path = os.path.join(tmpdir, 'm.dmatrix')
+            data = pd.DataFrame({
+                "a": [0, 1],
+                "b": [2, 3],
+                "c": [4, 5]
+            })
+            m0 = xgb.DMatrix(data.loc[:, ["a", "b"]], data["c"])
```

Comment:
```suggestion
            m0 = xgb.DMatrix(data.loc[:, ["a", "b"]], data["c"])
            assert m0.feature_names == ['a', 'b']
```
We should also check that the feature names of `m0` are not generic: